File: testdata/jennies/rawtypes/nullable_fields/JavaRawTypes/nullable_fields/NullableFieldsStructF.java
package nullable_fields;

import java.util.Objects;

public class NullableFieldsStructF {
    public String a;
    public NullableFieldsStructF() {
        this.a = "";
    }
    public NullableFieldsStructF(String a) {
        this.a = a;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof NullableFieldsStructF)) return false;
        NullableFieldsStructF o = (NullableFieldsStructF) other;
        if (!Objects.equals(this.a, o.a)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.a);
    }
}

File: testdata/jennies/rawtypes/nullable_fields/PHPRawTypes/src/NullableFields/NullableFieldsStructF.php
<?php

namespace Grafana\Foundation\NullableFields;

class NullableFieldsStructF implements \JsonSerializable
{
    public string $a;

    /**
     * @param string|null $a
     */
    public function __construct(?string $a = null)
    {
        $this->a = $a ?: "";
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->a !== $other->a) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{a?: string} $inputData */
        $data = $inputData;
        return new self(
            a: $data["a"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->a = $this->a;
        return $data;
    }
}
